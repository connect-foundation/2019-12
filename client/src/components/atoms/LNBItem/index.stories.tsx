import React from 'react';

import LNBItem from '.';

export default {
  title: 'Atoms / LNBItem',
};

export const index: React.FC = () => {
  return (
    <div style={{ display: 'flex' }}>
      <LNBItem children={'ë‚´ í‹°ì¼“'} />
    </div>
  );
};
