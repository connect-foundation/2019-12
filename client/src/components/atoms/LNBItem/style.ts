import styled from 'styled-components';
import { ifProp, palette, theme } from 'styled-tools';

interface ContainerProps {
  active: boolean;
}

export const Container = styled.div<ContainerProps>`
  ${theme('fontStyle.h6')};
  border-bottom: 0.3rem solid
    ${ifProp('active', palette('grayscale', 1), palette('white'))};
  color: ${palette('grayscale', 1)};
  padding: 2rem;
  font-weight: bold;
  transition: background-color ease 0.2s;
  user-select: none;
  cursor: pointer;
  flex: 1 1 0;
  text-align: center;

  &:hover {
    background-color: ${palette('grayscale', 6)};
  }
`;
