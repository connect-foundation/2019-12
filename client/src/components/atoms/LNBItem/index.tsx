import React, { useState } from 'react';
import * as S from './style';

export interface Props {
  children: string | React.ReactElement;
  active?: boolean;
}

function LNBItem({ children, active = false }: Props): React.ReactElement {
  const [isActive, setIsActive] = useState(active);

  return (
    <S.Container
      active={isActive}
      onClick={() => {
        !isActive && setIsActive(true);
      }}
    >
      {children}
    </S.Container>
  );
}

export default LNBItem;
