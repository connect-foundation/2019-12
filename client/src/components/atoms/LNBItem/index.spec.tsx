import React from 'react';
import { mount } from 'enzyme';
import LNBItem from '.';

describe('Atom / LNBItem', () => {
  it('[SNAPSHOT] 렌더링', () => {
    // given
    const wrapper = mount(<LNBItem active children={'내 티켓'} />);

    // then
    expect(wrapper).toMatchSnapshot();
  });
});
