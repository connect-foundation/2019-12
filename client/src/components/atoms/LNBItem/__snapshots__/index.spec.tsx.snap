// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom / LNBItem [SNAPSHOT] 렌더링 1`] = `
.c0 {
  border-bottom: 0.3rem solid;
  padding: 2rem;
  font-weight: bold;
  -webkit-transition: background-color ease 0.2s;
  transition: background-color ease 0.2s;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
  text-align: center;
}

<LNBItem
  active={true}
>
  <styled.div
    active={true}
    onClick={[Function]}
  >
    <StyledComponent
      active={true}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              ";
  border-bottom: 0.3rem solid
    ",
              [Function],
              ";
  color: ",
              [Function],
              ";
  padding: 2rem;
  font-weight: bold;
  transition: background-color ease 0.2s;
  user-select: none;
  cursor: pointer;
  flex: 1 1 0;
  text-align: center;

  &:hover {
    background-color: ",
              [Function],
              ";
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
    >
      <div
        className="c0"
        onClick={[Function]}
      >
        내 티켓
      </div>
    </StyledComponent>
  </styled.div>
</LNBItem>
`;
